File: search-algorithms/sequential-search/linear-search/js/linear-search.js
/*
worst case timing complexity = O(N)
*/

function linearSearch(array, size, wanted) {
  for (i = 0; i < size; i++) {
    if (array[i] === wanted) return i;
  }

  return -1;
}

const array = [20, 40, 23, 22, 90, 12];

console.log(linearSearch(array, array.length, 90));

File: search-algorithms/linear-search/python/linear-search.py
"""
worst case timing complexity = O(N)
"""

def linearSearch(vector, wanted):
    for i, v in enumerate(vector):
        if v == wanted: return i

    return -1

vector = [20, 21, 43, 2, 12, 122]

print(linearSearch(vector, 12))